src/tests/suite/daos_obj.c

```diff
@@ -3365,6 +3365,161 @@ io_pool_map_refresh_trigger(void **state)
 	ioreq_fini(&req);
 }
 
+/**
+ * Test to fetch the non existance keys.
+ */
+static void nonexistent_internal(void **state, daos_obj_id_t oid,
+				 const char key_type[])
+{
+	test_arg_t	*arg = *state;
+	struct ioreq	req;
+	char		*update_buf;
+	char		*fetch_buf;
+	unsigned int	size   = IO_SIZE_NVME;
+	const char	dkey[] = "nonexistent dkey";
+
+	D_ALLOC(update_buf, size);
+	assert_non_null(update_buf);
+	dts_buf_render(update_buf, size);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+	D_ALLOC(fetch_buf, size);
+	assert_non_null(fetch_buf);
+	memset(fetch_buf, 0, size);
+
+	/** create dkey to fetch nonexist akey*/
+	if (strcmp(key_type, "akey") == 0) {
+		insert_single(dkey, "first_akey", 0, update_buf, size,
+			      DAOS_TX_NONE, &req);
+	}
+
+	lookup_single(dkey, "nonexistent akey", 0, fetch_buf, size,
+		      DAOS_TX_NONE, &req);
+
+	/**
+	* As per current implementation non existing keys
+	* will not use the buffer at all during fetch
+	*/
+	assert_false(fetch_buf[0] != '\0');
+
+	D_FREE(update_buf);
+	D_FREE(fetch_buf);
+
+	ioreq_fini(&req);
+}
+
+/**
+ * Fetch the non existent dkey/akey.
+ */
+static void fetch_nonexistent_keys(void **state)
+{
+	test_arg_t	*arg = *state;
+	daos_obj_id_t	oid;
+	daos_handle_t	oh;
+	int		invalid_object = 100;
+	int		rc;
+
+	oid = dts_oid_gen(dts_obj_class, 0, arg->myrank);
+
+	/** Test non nonexistent dkey */
+	print_message("Fetch nonexistent DKEY\n");
+	nonexistent_internal(state, oid, "dkey");
+
+	/** Test non nonexistent akey */
+	print_message("Fetch nonexistent AKEY\n");
+	nonexistent_internal(state, oid, "akey");
+
+	/** Test nonexistent object type */
+	print_message("Fetch nonexistent OBJECT\n");
+	oid = dts_oid_gen(invalid_object, 0, arg->myrank);
+	rc  = daos_obj_open(arg->coh, oid, 0, &oh, NULL);
+	assert_int_not_equal(rc, 0);
```

Comment:
fetch is missed, so it's not actually testing fetch on non-existing object.

Reply:
ok so I need to find a way to open non-existing object before doing fetch.